src/ucp/wireup/wireup.c

```diff
@@ -1394,20 +1376,24 @@ ucp_wireup_check_is_reconfigurable(ucp_ep_h ep,
 {
     ucp_lane_index_t reuse_lane_map[UCP_MAX_LANES];
     const ucp_ep_config_key_t *old_key;
+    ucp_lane_index_t lane;
+
+    if (ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL) {
+        return 0;
+    }
 
     if (ucp_ep_has_cm_lane(ep)) {
         return 1;
     }
 
-    old_key = &ucp_ep_config(ep)->key;
-
-    /* Verify both old/new configurations have only p2p lanes */
-    if (!ucp_wireup_are_all_lanes_p2p(ep, old_key) ||
-        !ucp_wireup_are_all_lanes_p2p(ep, new_key) ||
-        (old_key->num_lanes != new_key->num_lanes)) {
-        return 0;
+    /* Verify configuration have only wired-up lanes */
+    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+        if (!ucp_wireup_ep_test(ucp_ep_get_lane(ep, lane))) {
```

Comment:
why must we have wireup_ep in order to be reconfigurable?

Reply:
because non-wireup eps require some more modifications which are done in subsequent PRs

Reply:
so pls update the comment